sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ReadingStandardInput.vb

```diff
@@ -0,0 +1,42 @@
+﻿Imports System
+Imports Con = System.Console
+
+Namespace Tests.Diagnostics
+    Class Program
+        Public Sub Method()
+            System.Console.Read() ' Noncompliant
+            Con.Read() ' Noncompliant
+
+            Console.Read() ' Noncompliant
+            Console.ReadKey() ' Noncompliant
+            Console.ReadKey(True) ' Noncompliant
+            Console.ReadLine() ' Noncompliant
+            Console.OpenStandardInput() ' Noncompliant
+            Console.OpenStandardInput(100) ' Noncompliant
+
+            Dim x = System.Console.[In] ' Noncompliant
+            x = Console.[In] ' Noncompliant
+            x = Con.[In] ' Noncompliant
+            Console.[In].Read() ' Noncompliant
+
+            ' Other console methods
+            Console.Write(1)
+            Console.WriteLine(1)
+            ' Other classes
+            MyConsole.Read()
+            MyConsole.[In].Read()
+        End Sub
+    End Class
+
+    Module MyConsole
```

Comment:
Are we matching the whole namespace, or just the containing type name i.e. would _MyNamespace.Console.Read_ give an FP?

Reply:
We are matching the fully qualified type name (but not the assembly name), so FPs are possible but that would be rare